docs/src/main/asciidoc/virtual-threads.adoc

```diff
@@ -185,313 +263,194 @@ public class FortuneResource {
+For example, with Maven, run:
 
-[source, java]
+[source, bash]
+----
+mvn package -Dnative
 ----
-//import ...
 
-@ApplicationScoped
-public class FortuneRepository {
-    // ...
+=== Using an in-container build
 
-    public Uni<List<Fortune>> findAllAsync() {
-        return db.getPool()
-                .preparedQuery(SELECT_ALL).execute()
-                .map(this::createListOfFortunes);
+In-container build allows building Linux 64 executables by using a `native-image` compiler running in a container.
+It avoids having to install `native-image` on your machine, and also allows configuring the GraalVM version you need.
+Note that, to use in-container build, you must have Docker or Podman installed on your machine.
 
-    }
+Then, add to your `application.properties` file:
 
-    public List<Fortune> findAllAsyncAndAwait() {
-        var rows = db.getPool().preparedQuery(SELECT_ALL)
-                .executeAndAwait();
-        return createListOfFortunes(rows);
-    }
+[source, properties]
+----
+quarkus.native.additional-build-args=--enable-preview <1>
 
-    //...
-}
+# In-container build to get a linux 64 executable
+quarkus.native.container-build=true <2>
+quarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel-builder-image:jdk-20 <3>
 ----
+<1> The `enable-preview` flag in only necessary until Java 21.
+<2> Enables the in-container build
+<3> The builder container to use. Make sure it supports virtual threads
 
-Contrary to the link:{pgsql-driver}[postgresql-jdbc driver], no `synchronized` block is used where it shouldn't be, and
-the `await` behaviour is implemented using locks and latches that won't cause pinning.
+[IMPORTANT]
+.From ARM/64 to AMD/64
+====
+If you are using a Mac M1 or M2 (using an ARM64 CPU), you need to be aware that the native executable you will get using an in-container build will be a Linux executable, but using your host (ARM 64) architecture.
+You can use emulation to force the architecture when using Docker with the following property:
 
-Using the synchronous methods of the link:{mutiny-vertx-sql}[smallrye-mutiny-vertx-sqlclient] along with virtual threads
-will allow you to use the synchronous blocking style, avoid pinning the carrier thread, and get performance close to a pure
-reactive implementation.
+[source, properties]
+----
+quarkus.native.container-runtime-options=--platform=linux/amd64
+----
 
-== A point about performance
+Be aware that it increases the compilation time... a lot (>10 minutes).
+====
 
-Our experiments seem to indicate that Quarkus with virtual threads will scale better than Quarkus blocking (offloading
-the computation on a pool of platform worker threads) but not as well as Quarkus reactive.
-The memory consumption especially might be an issue: if your system needs to keep its memory footprint low we would
-advise you stick to using reactive constructs.
+== Containerize applications using virtual threads
```

Comment:
Shall we move this section before [[native]] ? 
In this sequence it does native compilation, JVM container, native container. I think it would be easier to follow if it was JVM container, native compilation, native container.

Reply:
so splitting the containerization section into two distinguished section.

Reply:
When I was reading the containerization section I thought it was about containerization of native apps.

Reply:
Done